test/clj/cider/nrepl/main_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns cider.nrepl.main-test
+  (:require [cider.nrepl :refer [wrap-debug cider-middleware]]
+            [cider-nrepl.main :as m]
+            [clojure.test :refer :all]
+            [clojure.tools.nrepl :as nrepl]
+            [clojure.tools.nrepl.server :as nrepl.server]
+            [clojure.tools.nrepl.transport :as transport]))
+
+(defn start-stop-nrepl-session [opts]
```

Comment:
I don't see where you stop the server in each test.

Reply:
My understanding was that the `with-open` macro would suffice to close the server. I borrowed much of this from `cider.nrepl.middleware.debug-integration-test/with-nrepl-session*`

Reply:
Ah, ok.